### PR TITLE
Support collections of property-morprhable data

### PR DESCRIPTION
**Pull request by ncphillips**

### CODE REVIEW

File: src/Resolvers/DataValidationMessagesAndAttributesResolver.php [hunk 1]
```diff
@@ -74,15 +74,43 @@ public function execute(
...
+                        }
+
+                        $collected = $this->execute(
+                            $morphedClass,
+                            $fullPayload,
+                            $propertyPath->property($index),
+                            [...$nestingChain, $dataProperty->type->dataClass, $morphedClass],
+                        );
+
+                        $messages[] = $collected['messages'];
```
Comment (bentleyo) [inline src/Resolvers/DataValidationMessagesAndAttributesResolver.php:99]: If a morphed class cannot be determined for some items (the `continue` above) will these messages start being added to the wrong indexes?

I can't remember what the messages data structure looks like and I'm reviewing on my phone while on holidays so it's a bit tricky 😂 Just double checking if we need to do `$messages[$index] =` here.

File: tests/ValidationTest.php [hunk 2]
```diff
@@ -2549,11 +2549,11 @@ class TestDataWithMergedRuleset extends Data
     {
         public function __construct(
             #[Max(10)]
-            public string $array_rules,
+            public string                        $array_rules,
```
Comment (bentleyo) [inline tests/ValidationTest.php:2552]: Are these changes intentional?

File: tests/ValidationTest.php [hunk 3]
```diff
@@ -2699,7 +2699,9 @@ public function __construct(
                 public ?DataCollection $nestedCollection,
             ) {
             }
-        };
+        }
+
+        ;
```
Comment (bentleyo) [inline tests/ValidationTest.php]: This seems like a bit of an odd formatting change. Is this intentional?

**Review by bentleyo — CHANGES_REQUESTED** [review]
Thanks for this! Looks great, just added a comment with a question about an edge-case.

File: tests/ValidationTest.php (hunk 2 above)
Comment (ncphillips) [reply tests/ValidationTest.php:2552]: Nope, but I ran `composer format` so I assume it's their style guide

File: src/Resolvers/DataValidationMessagesAndAttributesResolver.php (hunk 1 above)
Comment (ncphillips) [reply src/Resolvers/DataValidationMessagesAndAttributesResolver.php:99]: Great catch! I'll add some more tests to see what happens

I'd actually say it should fall back to the base-class if the morphed one can't be found

Comment (ncphillips) [reply src/Resolvers/DataValidationMessagesAndAttributesResolver.php:99]: Okay, I confirmed it doesn't mess up the index, but it _is_ ignoring the base classes messages
image

File: tests/ValidationTest.php (hunk 3 above)
Comment (ncphillips) [reply tests/ValidationTest.php]: It must be that the semicolon is not required/desirable at the end of a class

**Review by bentleyo — APPROVED** [review]
